### PR TITLE
E2E: Product Query: Variations tests

### PR DESCRIPTION
**Pull request by dinhtungdu**
This PR adds E2E tests for Product Query variations including Products On Sale.

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests

#### User Facing Testing

Run test locally:
```
node bin/wp-env-with-gutenberg.js
GUTENBERG_EDITOR_CONTEXT=gutenberg npm run test:e2e tests/e2e/specs/backend/product-query/variations.test.ts
```

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

File: tests/e2e/config/jest.config.js [hunk 1]
```diff
@@ -32,4 +32,5 @@ module.exports = {
 	],
 	testPathIgnorePatterns: [ '<rootDir>/tests/e2e/specs/performance' ],
 	transformIgnorePatterns: [ 'node_modules/(?!(woocommerce)/)' ],
+	testMatch: [ '**/?(*.)+(spec|test).[jt]s?(x)' ],
```
Comment (dinhtungdu) [inline tests/e2e/config/jest.config.js:35]: Currently, we're using the config from `@wordpress/scripts`, which is:

```
	testMatch: [ '**/specs/**/*.[jt]s?(x)', '**/?(*.)spec.[jt]s?(x)' ],
```

I changed it to use one of Jest's [default match expressions](https://jestjs.io/docs/configuration#testmatch-arraystring). By doing so, we only match the test files with valid suffixes.

**Review by sunyatasattva — COMMENTED** [review]
Should we put this PR `on hold`? It looks like we might remove the variations next week…

**Comment by dinhtungdu** [comment]
> Should we put this PR `on hold`? It looks like we might remove the variations next week…

@sunyatasattva Sure, I just added `oh hold` label to this PR.

Can you take a look at other parts of this PR, beside the variation tests, I refactored the structure of other tests to make the code more reusable.

**Comment by sunyatasattva** [comment]
@dinhtungdu checked the other improvements and I like them very much! Test code is becoming cleaner and cleaner, thank you!

**Comment by dinhtungdu** [comment]
> @dinhtungdu checked the other improvements and I like them very much! Test code is becoming cleaner and cleaner, thank you!

Yay! I will move that part to another/new PR so we can merge them sooner.

**Comment by sunyatasattva** [comment]
Since you have moved the useful changes from this, and we have decided (p1670535184371009-slack-C02FL3X7KR6) to remove the variations, we can get rid of the rest of the code.

**Pull request closed without merging**